### ISSUE TITLE
Improve the format output with details about what needs to be formatted.

### ISSUE THREAD

**Issue by vlkorsakov** [astral-sh/ruff#9765]
Please provide a more detailed output, including the line number and a description of what I need to format as in flake8. I think you can update the output for use with the --check flag only.

**code example:**
```python
def foo():
    pass
def bar():
    pass
```

**flake8 output:**
```shell
my_awesome_project/main.py:3:1: E302 expected 2 blank lines, found 0
my_awesome_project/main.py:4:9: W292 no newline at end of file
```

**ruff format --check output:**
```shell
Would reformat: my_awesome_project/main.py
1 file would be reformatted
```

**Comment by zanieb** [comment]
Sounds like you're looking for #7352 with `ruff check`

Note that the formatter and linter are different, but if you want to see the format changes, run `ruff format --diff`.

**Comment by vlkorsakov** [comment]
> run `ruff format --diff`

This gives such a complex result. flake8's output is easier to understand. I think if you add more colours to diff (like in git diff), the output might become easier.

In general, I would like the result to be the same as "check" only because part of the pycodestyle rules are not supported right now (waiting for #9266), and I can only see what I missed by using the formatter. Thanks for the answer.

**Comment by zanieb** [comment]
You can get colors with something like `ruff format --diff | ydiff` (there are a bunch of different tools out there for viewing diffs). I agree that's not ideal though we should probably emit colored diffs by default!

> In general, I would like the result to be the same as "check"

Yeah I agree this would be nice if it were feasible.

**Comment by MichaReiser** [comment]
We plan to add color support to our diffs. See https://github.com/astral-sh/ruff/issues/9984 I'll close this in favour of that issue.
